pkg/skaffold/docker/image.go

```diff
@@ -179,9 +184,18 @@ func (l *localDaemon) Build(ctx context.Context, out io.Writer, workspace string
 		return "", fmt.Errorf("unable to evaluate build args: %w", err)
 	}
 
+	// Collect the images that the build will have to pull in order to minimize the number
+	// of authConfigs we have to compute and pass to the engine.
+	fromImages, err := fromImages(absDockerfilePath, buildArgs)
+
+	var authConfigs map[string]types.AuthConfig
 	// Like `docker build`, we ignore the errors
 	// See https://github.com/docker/cli/blob/75c1bb1f33d7cedbaf48404597d5bf9818199480/cli/command/image/build.go#L364
-	authConfigs, _ := DefaultAuthHelper.GetAllAuthConfigs(ctx)
+	if err != nil {
+		authConfigs, _ = DefaultAuthHelper.GetAllAuthConfigs(ctx)
+	} else {
+		authConfigs, _ = DefaultAuthHelper.GetSelectAuthConfigs(ctx, fromImages)
+	}
```

Comment:
Hi @dgageot ,  i think adding GetSelectAuthConfigs is not enough, if dev in local with docker-desktop or minikube in very early stage which push is not need, we don't need to check auth config.

see this issue https://github.com/GoogleContainerTools/skaffold/issues/4978

Reply:
hmm, @dfang that's a good point, but i think there are still cases when we still need to resolve auth right? if any external images need to be pulled then skipping auth would fail.

Reply:
yes, i got what you mean, private images.